test_scripts/Smoke/API/024_SubscribeVehicleData_PositiveCase.lua

```diff
@@ -0,0 +1,136 @@
+-- SDL receives VehicleInfo part of response from HMI with "SUCCESS" result code
+-- SDL responds with (resultCode: SUCCESS, success:true) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonSmoke = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Local Variables ]]
+local VDValues = {
+  gps = "VEHICLEDATA_GPS",
+  speed = "VEHICLEDATA_SPEED",
+  rpm = "VEHICLEDATA_RPM",
+  fuelLevel = "VEHICLEDATA_FUELLEVEL",
+  fuelLevel_State = "VEHICLEDATA_FUELLEVEL_STATE",
+  instantFuelConsumption = "VEHICLEDATA_FUELCONSUMPTION",
+  externalTemperature = "VEHICLEDATA_EXTERNTEMP",
+  prndl = "VEHICLEDATA_PRNDL",
+  tirePressure = "VEHICLEDATA_TIREPRESSURE",
+  odometer = "VEHICLEDATA_ODOMETER",
+  beltStatus = "VEHICLEDATA_BELTSTATUS",
+  bodyInformation = "VEHICLEDATA_BODYINFO",
+  deviceStatus = "VEHICLEDATA_DEVICESTATUS",
+  driverBraking = "VEHICLEDATA_BRAKING",
+  wiperStatus = "VEHICLEDATA_WIPERSTATUS",
+  headLampStatus = "VEHICLEDATA_HEADLAMPSTATUS",
+  engineTorque = "VEHICLEDATA_ENGINETORQUE",
+  accPedalPosition = "VEHICLEDATA_ACCPEDAL",
+  steeringWheelAngle = "VEHICLEDATA_STEERINGWHEEL",
+  eCallInfo = "VEHICLEDATA_ECALLINFO",
+  airbagStatus = "VEHICLEDATA_AIRBAGSTATUS",
+  emergencyEvent = "VEHICLEDATA_EMERGENCYEVENT",
+  clusterModeStatus = "VEHICLEDATA_CLUSTERMODESTATUS",
+  myKey="VEHICLEDATA_MYKEY"
+}
+
+local paramsList = {}
+local requestParams = { }
+for k, _ in pairs(VDValues) do
+  table.insert(paramsList, k)
+  requestParams[k] = true
+end
+
+local responseUiParams = { }
+
+local allParams = {
+  requestParams = requestParams,
+  responseUiParams = responseUiParams,
+}
+
+--[[ Local Functions ]]
+local function ptuUpdateFunc(tbl)
+  local SVDgroup = {
+    rpcs = {
+      SubscribeVehicleData = {
+        hmi_levels = { "BACKGROUND", "FULL", "LIMITED" },
+        parameters = paramsList
+      }
+    }
+  }
+  tbl.policy_table.functional_groupings.NewTestCaseGroup = SVDgroup
+  tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.appID].groups =
+  { "Base-4", "NewTestCaseGroup" }
+end
+
+local function setVDResponse()
+  local temp = { }
+  local vehicleDataResultCodeValue = "SUCCESS"
+  for k, v in pairs(VDValues) do
+    if "clusterModeStatus" == k then
+      temp["clusterModes"] = {
+        resultCode = vehicleDataResultCodeValue,
+        dataType = v
+      }
+    else
+      temp[k] = {
+        resultCode = vehicleDataResultCodeValue,
+        dataType = v
+      }
+    end
+  end
+  return temp
+end
+
+local function subscribeVD(pParams, self)
+  local cid = self.mobileSession1:SendRPC("SubscribeVehicleData", pParams.requestParams)
+  pParams.responseUiParams = setVDResponse()
+  EXPECT_HMICALL("VehicleInfo.SubscribeVehicleData", pParams.requestParams)
+  :Do(function(_,data)
```

Comment:
@GetmanetsIrina Please use space between , and `data`

Reply:
@dboltovskyi , Done in 7f6131f5913599c3d626af023f80827f9349e80a